corehq/apps/users/views/mobile/users.py

```diff
@@ -1065,20 +1066,35 @@ def post(self, request, *args, **kwargs):
             messages.error(request, _(str(e)))
             return HttpResponseRedirect(reverse(UploadCommCareUsers.urlname, args=[self.domain]))
 
-        upload_record = UserUploadRecord(
-            domain=self.domain,
-            user_id=request.couch_user.user_id
-        )
-        upload_record.save()
-
         task_ref = expose_cached_download(payload=None, expiry=1 * 60 * 60, file_extension=None)
-        task = import_users_and_groups.delay(
-            self.domain,
-            list(self.user_specs),
-            list(self.group_specs),
-            request.couch_user,
-            upload_record.pk
-        )
+        if PARALLEL_USER_IMPORTS.enabled(self.domain):
+            if list(self.group_specs):
+                messages.error(
+                    request,
+                    _("Groups are not allowed with parallel user import. Please upload them separately")
+                )
+                return HttpResponseRedirect(reverse(UploadCommCareUsers.urlname, args=[self.domain]))
+
+            task = parallel_user_import.delay(
+                self.domain,
+                list(self.user_specs),
```

Comment:
Are you concerned about the size of this param?  If the use-case is that very large imports are liable to fail or take forever, this might also be worth looking at (I recognize that it was already this way, so if it's not an issue, no need to change).  Wondering whether it'd be possible to push the spreadsheet parsing into the task or do the chunking here in some way.

Reply:
It was mostly speed, rather than failures, that inspired this, but it is interesting question if we can do this in chunks in the top level task, and parcel out the pieces rather than pass the full list. If this becomes a real feature (and not a one time thing for this nasty migration), it is definitely worth looking into.

Reply:
+1 to moving this into the task, even for normal user imports.